runtime/lua/vim/treesitter/_range.lua

```diff
@@ -25,52 +27,39 @@ local M = {}
 
 ---@alias Range Range2|Range4|Range6
 
----@param a_row integer
----@param a_col integer
----@param b_row integer
----@param b_col integer
----@return integer
---- 1: a > b
---- 0: a == b
---- -1: a < b
-local function cmp_pos(a_row, a_col, b_row, b_col)
-  if a_row == b_row then
-    if a_col > b_col then
-      return 1
-    elseif a_col < b_col then
-      return -1
-    else
-      return 0
-    end
-  elseif a_row > b_row then
-    return 1
-  end
-
-  return -1
-end
-
 M.cmp_pos = {
-  lt = function(...)
-    return cmp_pos(...) == -1
+  lt = function(a_row, a_col, b_row, b_col)
+    return pos(a_row, a_col) < pos(b_row, b_col)
```

Comment:
This module specifically uses arrays for performance reasons as treesitter needs to do lots of processing for injections.

This code now invokes a constructor twice, allocating around 4 tables just to do a comparison. Before it was just comparing 4 numbers.